lib/MemoryModel/MemModel.cpp

```diff
@@ -345,6 +345,12 @@ bool LocSymTableInfo::computeGepOffset(const User *V, LocationSet& ls)
                 Size_t num = at->getNumElements();
                 ls.addElemNumStridePair(std::make_pair(num, sz));
             }
+            else if (const StructType *ST = SVFUtil::dyn_cast<StructType>(*gi) )
```

Comment:
Will this ever be executed if this is a non-constant offset? Not quite sure, since it is bit a long time ago writing this.

Reply:
Yes in the case a gep indexes a structure that contains an array with a variable index like
```c
typedef struct A_s {
	int a;
	int arr[128];
} A_t;

printf("%d\n", s->arr[i]);
```
which translates to:
```c
%struct.A_s = type { i32, [128 x i32] }

%arrayidx = getelementptr inbounds %struct.A_s, %struct.A_s* %s, i64 0, i32 1, i64 %i
```

Reply:
The problem is that the branch is executed for every `index <= baseIndex`, is it necessary?